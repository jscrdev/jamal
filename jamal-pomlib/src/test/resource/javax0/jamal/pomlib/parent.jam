{@import res:javax0/jamal/pomlib/pom.jim}
deine the group, artifact and version
{GAV :grp:artfct:1.0.7}
use the group, artifact and version that was defined in the GAV
{parent :::}
{parent } no argument to parent at all, makes not sense, but for test it is checked

redefine group
{parent :parentGroupId::}

define artifact
{parent ::parentArtifactId:}

define version
{parent :::parentVersion}

define group and artifact
{parent :parentGroupId:parentArtifactId:}

define artifact and version
{parent ::parentArtifactId:parentVersion}

define group and version
{parent :parentGroupId::parentVersion}

define all
{parent :parentGroupId:parentArtifactId:parentVersion}
