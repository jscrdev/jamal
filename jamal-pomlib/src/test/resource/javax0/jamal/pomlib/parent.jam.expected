
deine the group, artifact and version

    <groupId>grp</groupId>
    <artifactId>artfct</artifactId>
    <version>1.0.7</version>

use the group, artifact and version that was defined in the GAV
<parent>
    <groupId>grp</groupId>
    <artifactId>artfct</artifactId>
    <version>1.0.7</version>
</parent>
<parent>
    <groupId>grp</groupId>
    <artifactId>artfct</artifactId>
    <version>1.0.7</version>
</parent> no argument to parent at all, makes not sense, but for test it is checked

redefine group
<parent>
    <groupId>parentGroupId</groupId>
    <artifactId>artfct</artifactId>
    <version>1.0.7</version>
</parent>

define artifact
<parent>
    <groupId>grp</groupId>
    <artifactId>parentArtifactId</artifactId>
    <version>1.0.7</version>
</parent>

define version
<parent>
    <groupId>grp</groupId>
    <artifactId>artfct</artifactId>
    <version>parentVersion</version>
</parent>

define group and artifact
<parent>
    <groupId>parentGroupId</groupId>
    <artifactId>parentArtifactId</artifactId>
    <version>1.0.7</version>
</parent>

define artifact and version
<parent>
    <groupId>grp</groupId>
    <artifactId>parentArtifactId</artifactId>
    <version>parentVersion</version>
</parent>

define group and version
<parent>
    <groupId>parentGroupId</groupId>
    <artifactId>artfct</artifactId>
    <version>parentVersion</version>
</parent>

define all
<parent>
    <groupId>parentGroupId</groupId>
    <artifactId>parentArtifactId</artifactId>
    <version>parentVersion</version>
</parent>